src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final int GROWTH_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] mapArray;
+
+    public MyHashMap() {
+        this.mapArray = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node newNode = new Node<>(key, value, null);
+        insert(index, newNode, key, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node oldNode = mapArray[getIndex(key)];
+        if (oldNode == null) {
+            return null;
+        }
+        while (true) {
+            if (isEqual((K) oldNode.key, key)) {
+                return (V) oldNode.value;
+            } else {
+                if (oldNode.next == null) {
+                    return null;
+                } else {
+                    oldNode = oldNode.next;
+                }
+            }
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resizeIfFull() {
+        if (mapArray.length * DEFAULT_LOAD_FACTOR == size) {
```

Comment:
what if resize was already done once?

you should replace DEFAULT_LOAD_FACTOR with some other value